src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java

```diff
@@ -96,17 +99,31 @@ public ParameterDefinition getParameterDefinition(String name) {
     }
 
     public Set<String> getUsersAsSet() {
+        return getUsersAsSet(null);
+    }
+
+    public Set<String> getUsersAsSet(AbstractBuild<?,?> build) {
         if (users == null || users.equals(""))
             return Collections.emptySet();
 
         Set<String> set = new HashSet<String>();
         for (String user : users.split(",")) {
             user = user.trim();
 
-            if (user.trim().length() > 0)
+            LOGGER.log(Level.FINER, "ManualCondition.getUsersAsSet(): user={0}", user);
+            if (user.startsWith("$")){
```

Comment:
You still need to null-check build somewhere before using it below. I think it makes sense to do it here: `if (build != null && user.startsWith("$"))`. That way if the the build is null, we don't try to substitute parameterized names, and just treat them as actual usernames.